src/main/java/org/kainos/ea/api/AuthService.java

```diff
@@ -0,0 +1,66 @@
+package org.kainos.ea.api;
+
+
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.security.Keys;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.kainos.ea.cli.LoginRequest;
+import org.kainos.ea.cli.User;
+import org.kainos.ea.client.FailedLoginException;
+import org.kainos.ea.client.FailedToGenerateTokenException;
+import org.kainos.ea.db.AuthDao;
+
+import java.security.Key;
+import java.sql.SQLException;
+import java.util.Date;
+
+public class AuthService {
+    private static final long EXPIRATION_TIME = 86400000; // 24 hours in milliseconds
+
+
+    public AuthService(AuthDao authDao) {
+        this.authDao = authDao;
+
+    }
+
+    private AuthDao authDao;
+
+
+
+
+    public String login(LoginRequest login) throws FailedLoginException {
+        try {
+            User user = authDao.validLogin(login);
+            if (user == null) {
+                return null;
+            }
+
+            String token = generateToken(user.getEmail());
+            if (token != null) {
+                return token;
+            }
+        } catch (SQLException e) {
+            throw new FailedToGenerateTokenException();
```

Comment:
This exception is missing from the method signature and isn't dealt with in the controller